### DISCUSSION TITLE
[Suggestion] Add some SkyBlock Island Information

### DISCUSSION THREAD

**Discussion by doej1367** [HypixelDev/PublicAPI#306] [Ideas]
There are multple numbers and details about a SkyBlock Profile Island that would be cool to see in the API:

- An **array with all the Minions that are placed down** 
(just type and level; maybe also fuel, hopper, items produced, the two upgrades and if they are boosted by their crystal)

- Some numbers about how many of a block-type there is on this Island 
(an Island-download - max. 16.3 MB + chest content - would probably be too much data for an api to handle)

  - an **int array with 255 fields, one for each of the main minecraft block ids, that contains all the block_counts** 
e.g. 154 (hopper, is its own id) or 1:3 (diorite falls under 1=stone because its a subcategory
(this is now a max szise of 4B * 255 = 1020B, wich is much more reasonable for an api, 
max int value for a block type would be 6,528,000 blocks = 10 chunks)

**Comment by xnacly** [comment]
id say the first point is very intresting, i would like to see that in the api.

**Comment by puppy0cam** [comment]
The API serves data directly from the database. Additional processing is not done on the data.

**Comment by doej1367** [comment]
> The API serves data directly from the database. Additional processing is not done on the data.

Guess you are refering to the second part that needs the calculation of the blocktype sum.
welp no building stats then... sadly.
The first one shouldn't be a problem though, becasue minions should be in the data base, aren' they?

**Comment by puppy0cam** [comment]
Most likely no. the minions placed down would be stored as part of the island data, as the offline calculation of resources is done when you enter the island.

**Comment by doej1367** [comment]
> Most likely no. the minions placed down would be stored as part of the island data, as the offline calculation of resources is done when you enter the island.

so you'd have to rework the whole database, and thats sadly not gonna happen...
thanks for clarifying 👍

**Comment by TheNullicorn** [comment]
I mean, theoretically they could add a heavily throttled endpoint for downloading an island's `.slime` file, and then allow players to opt into/out of making their island download public, but that would probably be way more trouble than it's worth (plus they would have to maintain a public version of the format spec, which I'm sure is past v3 by now).

**Comment by AzureAaron** [comment]
It would be amazing if we were able to see active minions, along with their appropriate levels & upgrades applied to them.

As for island downloads, not sure how this would honestly work, they could just let players download only their own personal islands as I can’t imagine anyone for the matter making an API viewer for this.

**Comment by doej1367** [comment]
> As for island downloads, not sure how this would honestly work, they could just let players download only their own personal islands as I can’t imagine anyone for the matter making an API viewer for this.

ye, thats why I mentioned the block statistic as an alterative.
The use of that would be, that you could calculate all the farms and an overall building efford out of these statistics,
Determaning a leaderbord of e.g. the biggest cactus farm, or the island with the most placed blocks or other cool things like it.
This would make it easier to find all the insane builders that the community has...

**Comment by MartinNemi03** [comment]
This could be revived with additional values for their island's name, parkours and parkour leaderboards, people's island points and social display leaderboards, easter egg count etc. Maybe even including #324 with Social XP.
